src/property-observation.js

```diff
@@ -96,7 +96,7 @@ export class SetterObserver {
 
     if (!Reflect.defineProperty(this.obj, this.propertyName, {
       configurable: true,
-      enumerable: true,
+      enumerable: this.obj.propertyIsEnumerable(this.propertyName),
```

Comment:
Right- there's no guarantee the property will be defined so the change you mentioned ^^^ would be necessary.  Could you add a test for that scenario as well?  Otherwise this looks great!